### ISSUE TITLE
Make script blocking UI self-explanatory

### ISSUE THREAD

**Issue by Hoeze** [uBlockOrigin/uBlock-issues#2548]
### Prerequisites

- [X] I verified that this is not a filter list issue. Report any issues with filter lists or broken website functionality in the [uAssets issue tracker](https://github.com/uBlockOrigin/uAssets/issues).
- [X] This is not a support issue or a question. For support, questions, or help, visit [/r/uBlockOrigin](https://www.reddit.com/r/uBlockOrigin/).
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uBlock-issues/issues?q=is%3Aissue) to avoid opening a duplicate issue.
- [X] The issue is not present after disabling uBO in the browser.
- [X] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I am reporting is not normal behavior.

### I tried to reproduce the issue when...

- [X] uBO is the only extension.
- [X] uBO uses default lists and settings.
- [X] using a new, unmodified browser profile.

### Description

I just found that uBO has a script blocking feature which could make NoScript obsolete for me.
However, I just invested at least one hour into reading through the uBlock documentation and some guides to find out how it works.

Please make this a self-explanatory feature, at least for people who are used to NoScript.
Especially some tool tips that explain each field would help MASSIVELY in understanding what I am about to do.

Here an example how this looks for NoScript:
![image](https://user-images.githubusercontent.com/1200058/225326249-fb67ce7b-2f37-4cce-b701-cf936d9e10c4.png)
The same for uBO:
![image](https://user-images.githubusercontent.com/1200058/225326514-90864771-47ce-4c14-985c-e7e355cd9f78.png)
- What do the two color columns mean?
- What means red / grey?
- What do the `+`/`++` signs mean?

Please add at least some tooltips: `[globally] block/whitelist javascript on domain`.
This would have already covered all I wanted to do and saved me tons of time reading guides on uBO.

Also, please do not mention "filter mode" everywhere in the wiki and not have a configuration option called "filter mode".
I wasted quite some time to find somewhere an "filter mode" option which just does not exist.

### A specific URL where the issue occurs.

not relevant, example: github.com

### Steps to Reproduce

- Open uBO as a new user
- Try to find out how script blocking works

### Expected behavior

- Enable advanced mode
- See new columns in the uBO UI
- Hover over them to find out that they are there for script blocking

### Actual behavior

- Be confused where is the script blocking now
- Read through github Wiki just to be more confused because there are no screenshots on dynamic filtering
- Find out that one needs to setup "medium filtering mode"
- Search for "filter mode" in uBlock UI
- Give up because there is no such "filter mode" configuration field
- Search google on how the fck to use uBlock for script filtering
- Reading four different guides until this one finally explains what I want to know:
  https://modernprivatelife.com/beginners-guide-to-ublock-origin/

### uBO version

1.47.4

### Browser name and version

Firefox 111.0

### Operating System and version

Arch Linux

**Comment by garry-ut99** [comment]
> Read through github Wiki just to be more confused because there are no screenshots on dynamic filtering
> Reading four different guides until this one finally explains what I want to know:
[modernprivatelife.com/beginners-guide-to-ublock-origin](https://modernprivatelife.com/beginners-guide-to-ublock-origin/)

Does that mean that you missed this: https://github.com/uBlockOrigin/uBlock-issues/wiki/Quick-guide:-popup-user-interface#the-overview-panel

**Comment by gorhill** [comment]
Also, _"make script blocking UI self-explanatory"_ will just cause the UI to become cluttered and unpleasant to use, so I rather not make any change here, that UI has been in effect for years now.

There are tooltips for the top cell of each column to remind you the purpose, and there are a lot of visual hints that serve as reminder of which column is global and which is local: the global rules propagate to local rules but not the other way around, the pluses/minuses reflect the blocked/allowed network requests, which are always reported in the local-rule column.

People who are used to uBO's approach to blocking scripts could also complain about NoScript's approach -- in the end whoever is accustomed to one approach of working could complain that the other is deficient, that doesn't mean that the other has to be changed, it's just a matter of getting accustomed.

**Comment by Hoeze** [comment]
Thanks @gorhill for your explanation.
> There are tooltips for the top cell of each column to remind you the purpose, and there are a lot of visual hints that serve as reminder of which column is global and which is local: the global rules propagate to local rules but not the other way around, the pluses/minuses reflect the blocked/allowed network requests, which are always reported in the local-rule column.

Yes, indeed. I just never came to see the tooltip there.
Probably because I did not want to block /unblock ALL scripts.

Do you think it would be viable to have the tooltip on all cells and not only on the first row?
Also, would it be possible to have different tooltips for the different color cells?

> People who are used to uBO's approach to blocking scripts could also complain about NoScript's approach -- in the end whoever is accustomed to one approach of working could complain that the other is deficient, that doesn't mean that the other has to be changed, it's just a matter of getting accustomed.

I understand UI of uBO now, I was simply frustrated by the time it took me to understand how it works.
I wanted to share my journey here to maybe save new users a bit of my frustration.

**Comment by gorhill** [comment]
> would it be possible to have different tooltips for the different color cells?

See <https://github.com/gorhill/uBlock/issues/1391#issuecomment-184832937>.

Beside being quite an annoyance to have tooltips pop out each time we hover over any of the cells, no amount of tooltips will avoid having to minimally read the documentation, at least [the quick guide](https://github.com/gorhill/uBlock/wiki/Dynamic-filtering:-quick-guide) -- and there is a lot of reminders that the documentation should be read when enabling _I am an advanced user_.

![Screenshot from 2023-03-15 10-46-43](https://user-images.githubusercontent.com/585534/225346698-70f5f30e-631e-49f8-8053-025cf3b905d2.png)

Which lead to [Advanced user features](https://github.com/gorhill/uBlock/wiki/Advanced-user-features), which lead to [Dynamic filtering](https://github.com/gorhill/uBlock/wiki/Dynamic-filtering), which lead to [Dynamic filtering: quick guide](https://github.com/gorhill/uBlock/wiki/Dynamic-filtering:-quick-guide).

The _Blocking modes_ documentation regarding Medium mode you show in your comment above has a link to [Blocking mode: medium mode](https://github.com/gorhill/uBlock/wiki/Blocking-mode:-medium-mode), which contains this paragraph near the top (emphasis added):

> This is where you start to use [dynamic filtering](https://github.com/gorhill/uBlock/wiki/Dynamic-filtering), a feature available only when you tell uBlock Origin (uBO) that you are an [advanced user](https://github.com/gorhill/uBlock/wiki/Advanced-user-features). **Be sure to read [the guide](https://github.com/gorhill/uBlock/wiki/Dynamic-filtering) before, it is assumed that you understand well how dynamic filtering works in order to use medium mode effectively.**

On that note, I will argue that NoScript is not that self explanatory, you can regularly see on NoScript forum people not understanding well how NoScript works (it also needs [its own documentation](https://noscript.net/usage/)) -- in the end when dealing with advanced features as script-blocking rules, there is a minimum of reading required and a minimual understanding of how the web work.

**Comment by garry-ut99** [comment]
> See https://github.com/gorhill/uBlock/issues/1391#issuecomment-184832937.

Also the others:

- https://github.com/gorhill/uBlock/issues/241
- https://github.com/gorhill/uBlock/issues/451
- https://github.com/gorhill/uBlock/issues/639
- https://github.com/gorhill/uBlock/issues/784
- https://github.com/gorhill/uBlock/issues/1069
- https://github.com/gorhill/uBlock/issues/2230
- https://github.com/uBlockOrigin/uBlock-issues/issues/20

Most likely the current issue is a duplicate of one of the above.
